cli/cmd/release_lint.go

```diff
@@ -89,9 +49,19 @@ func (r *runners) releaseLintStable(cmd *cobra.Command, args []string) error {
 		return errors.Errorf("fail-on value %q not supported, supported values are [info, warn, error, none]", r.args.lintReleaseFailOn)
 	}
 
-	lintReleaseYAML, err := readYAMLDir(r.args.lintReleaseYamlDir)
-	if err != nil {
-		return errors.Wrap(err, "read yaml dir")
+	var lintReleaseYAML []byte
+	if r.appType == "kots" {
+		b, err := tarYAMLDir(r.args.lintReleaseYamlDir)
+		if err != nil {
+			return errors.Wrap(err, "failed to read yaml dir")
+		}
+		lintReleaseYAML = b
+	} else {
```

Comment:
I don't really understand this -- for Platform, linting is not supported, and for ship there's no Dir, its just a single file

Reply:
This was very confusing for me as well, but i did not change any functionality. I simple eliminated the old route for kots linting.

Reply:
Check out https://github.com/replicatedhq/replicated/pull/132/files#diff-4a4b54d68a6f920d2d4433fa0a656336L92